api/v1alpha1/clusterextension_types.go

```diff
@@ -69,100 +68,93 @@ type ClusterExtensionSpec struct {
 	//    namespace: example-namespace
 	//    serviceAccount:
 	//      name: example-sa
+	//
+	// +kubebuilder:validation:Required
 	Install ClusterExtensionInstallConfig `json:"install"`
 }
 
 const SourceTypeCatalog = "Catalog"
 
 // SourceConfig is a discriminated union which selects the installation source.
+//
 // +union
-// +kubebuilder:validation:XValidation:rule="self.sourceType == 'Catalog' && has(self.catalog)",message="sourceType Catalog requires catalog field"
+// +kubebuilder:validation:XValidation:rule="has(self.sourceType) && self.sourceType == 'Catalog' ? has(self.catalog) : !has(self.catalog)",message="catalog is required when sourceType is Catalog, and forbidden otherwise"
 type SourceConfig struct {
 	// sourceType is a required reference to the type of install source.
 	//
-	// Allowed values are ["Catalog"]
+	// Allowed values are "Catalog"
 	//
-	// When this field is set to "Catalog", information for determining the appropriate
-	// bundle of content to install will be fetched from ClusterCatalog resources existing
-	// on the cluster. When using the Catalog sourceType, the catalog field must also be set.
+	// When this field is set to "Catalog", information for determining the
+	// appropriate bundle of content to install will be fetched from
+	// ClusterCatalog resources existing on the cluster.
+	// When using the Catalog sourceType, the catalog field must also be set.
 	//
 	// +unionDiscriminator
 	// +kubebuilder:validation:Enum:="Catalog"
+	// +kubebuilder:validation:Required
 	SourceType string `json:"sourceType"`
 
-	// catalog is used to configure how information is sourced from a catalog. This field must be defined when sourceType is set to "Catalog",
-	// and must be the only field defined for this sourceType.
+	// catalog is used to configure how information is sourced from a catalog.
+	// This field is required when sourceType is "Catalog", and forbidden otherwise.
 	//
 	// +optional.
 	Catalog *CatalogSource `json:"catalog,omitempty"`
 }
 
 // ClusterExtensionInstallConfig is a union which selects the clusterExtension installation config.
 // ClusterExtensionInstallConfig requires the namespace and serviceAccount which should be used for the installation of packages.
+//
 // +union
 type ClusterExtensionInstallConfig struct {
-	// namespace is a reference to the Namespace in which the bundle of
-	// content for the package referenced in the packageName field will be applied.
+	// namespace designates the kubernetes Namespace where bundle content
+	// for the 'packageName' package field will be applied and the necessary
```

Comment:
I know Joel was a bit confused by the previous verbiage, but now reading this I'm getting confused. Maybe something like:
```suggestion
	// for the package, referenced in the packageName field, will be applied and the necessary
```

Reply:
This was left half-done I think.  WDYT about something like
> namespace designates the kubernetes Namespace which will be used to install the package, and where the necessary service account can be found.

?